tools/integration_tests/integration_test_utils.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from __future__ import annotations
+
+import json
+import logging
+import subprocess
+import sys
+from pathlib import Path
+from typing import final, Sequence
+
+LOG: logging.Logger = logging.getLogger(__name__)
+
+
+@final
+class PyreErrorException(Exception):
+    """
+    Custom Exception to raise when Pyre errors out
+    """
+
+    pass
+
+
+def normalized_json_dump(
+    results: str, strip_locations: bool, filter_issues: bool
+) -> str:
+    """
+    Returns a normalised JSON string from results keeping only essential items. Filters
+    issues down to issues that have the code we intend to test for if filter_issues is
+    true, and removes location information if strip_location is set to True.
+    """
+    normalized = json.loads(results)
+    if "errors" in normalized:
+        pretty_error = json.dumps(normalized, sort_keys=True, indent=2)
+        raise PyreErrorException(
+            f"Errors were found when processing analysis results:\n{pretty_error}"
+        )
+
+    if filter_issues:
+        # Filter down to only issues that have the code that we intended to test for.
+        # This prevents the introduction of new rules or false positives from
+        # breaking existing tests.
+        normalized = [
+            issue for issue in normalized if f"test_{issue['code']}_" in issue["define"]
+        ]
+
+    normalized = sorted(
+        normalized,
+        key=lambda issue: (
+            issue["code"],
+            issue["path"],
+            issue["line"],
+            issue["column"],
+        ),
+    )
+
+    if strip_locations:
+        stripped_keys = {"column", "line", "stop_column", "stop_line"}
+        normalized = [
+            {key: value for key, value in issue.items() if key not in stripped_keys}
+            for issue in normalized
+        ]
+
+    return json.dumps(normalized, sort_keys=True, indent=2) + "\n"
+
+
+def friendly_exit(error_message: str, logs: str, suggested_hash: str) -> None:
+    """
+    Error function to print error message using LOG and exit
+    """
+    LOG.error("----BEGIN PYSA INTEGRATION TEST ERROR----")
+    LOG.error(error_message)
+    LOG.error(logs)
+    LOG.error("----END PYSA INTEGRATION TEST ERROR----")
+    sys.exit(1)
+
+
+def run_pysa_analysis(
```

Comment:
I would give it a name like `run_pysa_integration_test`. It does more than just running pysa.